contentcuration/contentcuration/frontend/accounts/pages/Create.vue

```diff
@@ -2,35 +2,428 @@
+        return [() => !!this.form.locations.length || this.$tr('fieldRequiredMessage')];
+      },
+      sources() {
+        return sources;
+      },
+      sourceOptions() {
+        return [
+          {
+            id: sources.ORGANIZATION,
+            label: this.$tr('organizationSourceOption'),
+            additional: {
+              model: this.form.organization,
+              label: this.$tr('organizationSourcePlaceholder'),
+            },
+          },
+          {
+            id: sources.WEBSITE,
+            label: this.$tr('websiteSourceOption'),
+          },
+          {
+            id: sources.NEWSLETTER,
+            label: this.$tr('newsletterSourceOption'),
+          },
+          {
+            id: sources.FORUM,
+            label: this.$tr('forumSourceOption'),
+          },
+          {
+            id: sources.GITHUB,
+            label: this.$tr('githubSourceOption'),
+          },
+          {
+            id: sources.SOCIAL_MEDIA,
+            label: this.$tr('socialMediaSourceOption'),
+          },
+          {
+            id: sources.CONFERENCE,
+            label: this.$tr('conferenceSourceOption'),
+            additional: {
+              model: this.form.conference,
+              label: this.$tr('conferenceSourcePlaceholder'),
+            },
+          },
+          {
+            id: sources.CONVERSATION,
+            label: this.$tr('conversationSourceOption'),
+          },
+          {
+            id: sources.DEMO,
+            label: this.$tr('personalDemoSourceOption'),
+          },
+          {
+            id: sources.OTHER,
+            label: this.$tr('otherSourceOption'),
+            additional: {
+              model: this.form.other_source,
+              label: this.$tr('otherSourcePlaceholder'),
+            },
+          },
+        ];
+      },
+      sourceRules() {
+        return [() => !!this.form.source.length || this.$tr('fieldRequiredMessage')];
+      },
+      policyLink() {
+        return window.Urls.policies();
+      },
+    },
+    beforeMount() {
+      this.form.email = this.$route.query.email || '';
     },
     methods: {
+      ...mapActions('account', ['register']),
+      showStorageField(id) {
+        return id === uses.STORING && this.form.uses.includes(id);
+      },
+      showOtherField(id) {
+        return id === uses.OTHER && this.form.uses.includes(id);
+      },
+      clean() {
+        Object.keys(this.form).forEach(key => {
+          // Trim text fields
+          if (typeof this.form[key] === 'string') {
+            this.form[key] = this.form[key].trim();
+          }
+        });
+      },
       submit() {
-        // handle account creation
-        this.$router.push({ name: 'ActivationSent' });
+        this.clean();
+        if (this.$refs.form.validate()) {
+          return this.register({
+            ...this.form,
+            uses: this.form.uses.join('|'),
+            locations: this.form.locations.join('|'),
+          })
```

Comment:
Writing methods like `.clean()` to imperatively modify vue data is often an anti-pattern in reactive systems, partly because these changes can propagate back to the UI in unexpected ways.

A more idiomatic way of doing this would be to write a computed prop like `processedFormData` which returns a copy of the data, after having trimmed strings and joined arrays with `|` characters.

Then you can write simply:

```js
  return this.register(this.processedFormData)
```